crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -2219,54 +2407,36 @@ where
     }
 }
 
-/// A [`Command`] that despawns a specific entity.
-/// This will emit a warning if the entity does not exist.
+/// An [`EntityCommand`] that despawns a specific entity.
 ///
 /// # Note
 ///
 /// This won't clean up external references to the entity (such as parent-child relationships
 /// if you're using `bevy_hierarchy`), which may leave the world in an invalid state.
 #[track_caller]
-fn despawn(log_warning: bool) -> impl EntityCommand {
+fn despawn() -> impl EntityCommand<World> {
+    #[cfg(feature = "track_change_detection")]
```

Comment:
This change seems correct regardless, maybe we should split this out? Ditto the docs above.

Reply:
I think that would need the `impl EntityCommand for impl FnOnce(EntityWorldMut)` changes. I do think that should be pulled out of this pr as that is something that could be considered separately.